File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0;
        int j = nums.size() -1;
        vector<pair<int, int>> pairs;

        for(int i = 0; i < nums.size(); i++){
            pairs.push_back({nums[i], i});
        }

        sort(pairs.begin(), pairs.end());

        while(i < j){
            int sum = pairs[i].first + pairs[j].first;
            if(sum == target){
                return {pairs[i].second, pairs[j].second};
            }else if(sum > target){
                j--;
            }else{
                i++;
            }

        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sorted_array = []
        for i, val in enumerate(nums):
            sorted_array.append((val, i))
        sorted_array.sort()

        n = len(nums)
        left = 0
        right = n-1

        while left < right:
            summ = sorted_array[left][0] + sorted_array[right][0]
            if summ == target:
                return [sorted_array[left][1], sorted_array[right][1]]

            if summ > target:
                right -= 1
            else:
                left += 1
        
        return []
            